anchor-service/src/time_event.rs

```diff
@@ -0,0 +1,162 @@
+use anyhow::{anyhow, Result};
+use cid::Cid;
+
+use ceramic_event::{
+    anchor::{AnchorRequest, TimeEvents},
+    unvalidated::RawTimeEvent,
+};
+
+use crate::DetachedTimeEvent;
+
+pub fn build_time_events(
+    anchor_requests: &[AnchorRequest],
+    detached_time_event: &DetachedTimeEvent,
+    count: u64,
+) -> Result<TimeEvents> {
+    let events = anchor_requests
+        .iter()
+        .enumerate()
+        .map(|(index, anchor_request)| {
+            build_time_event(
+                &anchor_request.id,
+                &anchor_request.prev,
+                &detached_time_event.proof,
+                detached_time_event.path.as_str(),
+                index.try_into()?,
+                count,
+            )
+        })
+        .collect::<Result<Vec<_>>>()?;
+
+    Ok(TimeEvents { events })
+}
+
+pub fn build_time_event(
```

Comment:
I wonder if it's worth adding an input struct here to make it slightly harder to mismatch all the cids.. it's still a bunch of cids but it was something like one of these two might be less error prone (if a bit tedious)

```rust
    struct BuildTimeEventInput<'a> {
        pub id: &'a Cid,
        pub prev: &'a Cid,
        pub root: &'a Cid,
        pub remote_path: &'a str,
        pub index: u64,
        pub count: u64,
    }

    struct BuildTimeEventInput2<'a> {
        pub id: Cid,
        pub prev: Cid,
        pub root: Cid,
        pub remote_path: &'a str,
        pub index: u64,
        pub count: u64,
    }

    fn build_it(input: BuildTimeEventInput) {
        todo!()
    }
    fn build_it2(input: &BuildTimeEventInput2) {
        todo!()
    }
    build_it(BuildTimeEventInput {
        id: &id,
        prev: &prev,
        root: &proof,
        remote_path: "",
        index,
        count,
    });
    // only need to assign input here if we still want access to the cids after
    let input = BuildTimeEventInput2 {
        id,
        prev,
        root: proof,
        remote_path: "",
        index,
        count,
    };

    build_it2(&input);
 ```

Reply:
We went ahead and inlined the function.